app/gui/view/welcome-screen/src/side_menu.rs

```diff
@@ -58,4 +88,36 @@ impl SideMenu {
         projects_list.append_or_warn(&button, logger);
         button
     }
+
+    fn setup_new_project_event_listener(&self, create_project: CreateProject) {
+        let closure = Box::new(move |_event: MouseEvent| {
+            create_project.emit(None);
+        });
```

Comment:
This is not the best usage of FRP. You should not really emit events from outside of FRP networks until this is something very special (like low-0level keyboard binding). Such things should be part of the FRP network. This is important – otherwise the code becomes spaghetti that is hard to track.

Reply:
Clicking this button should create FRP event, then this FRP event should be connected in the FRP network to do something. So basically, JS events such as "clicked this div" should just emit FRP `()` event, and all logic should be handled within FRP network instead.

Reply:
I don't understand this, because
> So basically, JS events such as "clicked this div" should just emit FRP () event, and all logic should be handled within FRP network instead.

describes the current implementation quite closely. `create_project` is a `frp::Source` which is connected to WelcomeScreen's FRP network output `create_project` by this code: https://github.com/enso-org/enso/pull/3167/files#diff-2876c5422d672917ce2f22754a119c7b32d17244afc7374989389ef2ded8c161R193-R197